src/core/lib/iomgr/resource_quota.cc

```diff
@@ -121,26 +123,45 @@ struct grpc_resource_user {
   char* name;
 };
 
-struct grpc_resource_quota {
-  /* refcount */
-  gpr_refcount refs;
+static void rq_step(void* rq, grpc_error* error);
+static void rq_reclamation_done(void* rq, grpc_error* error);
+
+struct grpc_resource_quota : grpc_core::IntrusivelyRefCounted<grpc_resource_quota> {
```

Comment:
If we're going to C++-ify this, let's do it all the way: make it a class, convert the standalone functions into methods, use C++ naming conventions, etc.

Reply:
I removed `grpc_core::IntrusivelyRefCounted`. This CL is just a demo of a much larger change that replaces `gpr_refcount` with a C++ class that is significantly more efficient. 

If you look at the original intent, there are 66 files changed in basically all modules:
https://github.com/grpc/grpc/pull/17124/files

Just to use this class I have to convert all of the structs in the 66 files to C++ classes with ctor/dtor similar to memset(0). Full C++ification would be great, but as separate clean up CLs IMHO.